flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/fetch/PostgresStreamFetchTask.java

```diff
@@ -213,24 +213,19 @@ public void execute(
 
                 LOG.debug("StreamSplit is bounded read: {}", streamSplit);
                 final PostgresOffset currentOffset = PostgresOffset.of(offsetContext.getOffset());
-
-                if (currentOffset.isAtOrAfter(streamSplit.getEndingOffset())) {
-                    LOG.info("StreamSplitReadTask finished for {}", streamSplit);
-
-                    try {
-                        dispatcher.dispatchWatermarkEvent(
-                                partition.getSourcePartition(),
-                                streamSplit,
-                                currentOffset,
-                                WatermarkKind.END);
-                    } catch (InterruptedException e) {
-                        LOG.error("Send signal event error.", e);
-                        errorHandler.setProducerThrowable(
-                                new FlinkRuntimeException("Error processing WAL signal event", e));
-                    }
-
-                    ((PostgresScanFetchTask.PostgresChangeEventSourceContext) context).finished();
+                try {
+                    dispatcher.dispatchWatermarkEvent(
```

Comment:
Add a log for troubleshooting like:

                    LOG.info("StreamSplitReadTask finished for {} at {}", streamSplit, currentOffset);